Race condition when creating the Opensearch client in the logging plugin

Looks like it will sometimes try to create the client before the credentials secret exists:
```
2022-12-16T00:18:56.427Z ERROR plugin.logging.opensearch-manager gateway/admin_v2.go:1002 failed to create client: secrets "opensearch-opni-internalopni" not found
panic: secrets "opensearch-opni-internalopni" not found
goroutine 282 [running]:
github.com/rancher/opni/plugins/logging/pkg/gateway.(*LoggingManagerV2).setOpensearchClient(0xc001086180)
github.com/rancher/opni/plugins/logging/pkg/gateway/admin_v2.go:1003 +0xc8d
github.com/rancher/opni/plugins/logging/pkg/util.(*AsyncOpensearchClient).SetClient(0xc000d8dbf0, 0xc000ae8360)
github.com/rancher/opni/plugins/logging/pkg/util/opensearch.go:41 +0xdc
created by github.com/rancher/opni/plugins/logging/pkg/gateway.Scheme
github.com/rancher/opni/plugins/logging/pkg/gateway/plugin.go:231 +0x3ea
2022-12-16T00:18:56.432Z [ERROR] plugin process exited: path=/var/lib/opni/plugins/plugin_logging pid=14 error="exit status 2"
```

--- a/pkg/opensearch/certs/k8s_cert_manager.go
+++ b/pkg/opensearch/certs/k8s_cert_manager.go
@@ -6,9 +6,11 @@ import (
 	"crypto/x509"
 	"errors"
 	"fmt"
+	"time"
 
 	cmv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
 	cmmetav1 "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	backoffv2 "github.com/lestrrat-go/backoff/v2"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
@@ -190,29 +192,59 @@ func (m *certMgrOpensearchManager) GetHTTPRootCAs() (*x509.CertPool, error) {
 }
 
 func (m *certMgrOpensearchManager) GetClientCert(user string) (tls.Certificate, error) {
+	p := backoffv2.Exponential(
+		backoffv2.WithMinInterval(time.Second*2),
+		backoffv2.WithMaxInterval(time.Second*5),
+		backoffv2.WithMaxRetries(5),
+		backoffv2.WithMultiplier(1.2),
+	)
+
 	cert := &corev1.Secret{}
-	err := m.k8sClient.Get(m.ctx, types.NamespacedName{
-		Name:      m.clientCertName(user),
-		Namespace: m.storageNamespace,
-	}, cert)
-	if err != nil {
-		return tls.Certificate{}, err
+	var err error
+	b := p.Start(m.ctx)
+	for backoffv2.Continue(b) {
+		err = m.k8sClient.Get(m.ctx, types.NamespacedName{
+			Name:      m.clientCertName(user),
+			Namespace: m.storageNamespace,
+		}, cert)
+		if err != nil {
+			if ctrlclient.IgnoreNotFound(err) != nil {
+				return tls.Certificate{}, err
+			}
+			continue
+		}
+		return tls.X509KeyPair(cert.Data[corev1.TLSCertKey], cert.Data[corev1.TLSPrivateKeyKey])
 	}
 
-	return tls.X509KeyPair(cert.Data[corev1.TLSCertKey], cert.Data[corev1.TLSPrivateKeyKey])
+	return tls.Certificate{}, err
 }
 
 func (m *certMgrOpensearchManager) GetAdminClientCert() (tls.Certificate, error) {
+	p := backoffv2.Exponential(
+		backoffv2.WithMinInterval(time.Second*2),
+		backoffv2.WithMaxInterval(time.Second*5),
+		backoffv2.WithMaxRetries(5),
+		backoffv2.WithMultiplier(1.2),
+	)
+
 	cert := &corev1.Secret{}
-	err := m.k8sClient.Get(m.ctx, types.NamespacedName{
-		Name:      m.adminCertName(),
-		Namespace: m.storageNamespace,
-	}, cert)
-	if err != nil {
-		return tls.Certificate{}, err
+	var err error
+	b := p.Start(m.ctx)
+	for backoffv2.Continue(b) {
+		err = m.k8sClient.Get(m.ctx, types.NamespacedName{
+			Name:      m.adminCertName(),
+			Namespace: m.storageNamespace,
+		}, cert)
+		if err != nil {
+			if ctrlclient.IgnoreNotFound(err) != nil {
+				return tls.Certificate{}, err
+			}
+			continue
+		}
+		return tls.X509KeyPair(cert.Data[corev1.TLSCertKey], cert.Data[corev1.TLSPrivateKeyKey])
 	}
 
-	return tls.X509KeyPair(cert.Data[corev1.TLSCertKey], cert.Data[corev1.TLSPrivateKeyKey])
+	return tls.Certificate{}, err
 }
 
 func (m *certMgrOpensearchManager) GetTransportCARef() (corev1.LocalObjectReference, error) {

